tigroo31/libflow
# Fix flow eq

fix #7 
fix #6 

**How to test:**

Run the project and check the unit tests pass:

```
Testing started at 14:56 ...
/home/kvmk8371/.cargo/bin/cargo test --color=always --no-fail-fast -- -Z unstable-options --format=json --show-output
warning: variant is never constructed: `FIN`
 --> src/flow/flag.rs:3:5
  |
3 |     FIN,
  |     ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: variant is never constructed: `PSH`
 --> src/flow/flag.rs:4:5
  |
4 |     PSH,
  |     ^^^

warning: variant is never constructed: `URG`
 --> src/flow/flag.rs:5:5
  |
5 |     URG,
  |     ^^^

warning: variant is never constructed: `ECE`
 --> src/flow/flag.rs:6:5
  |
6 |     ECE,
  |     ^^^

warning: variant is never constructed: `SYN`
 --> src/flow/flag.rs:7:5
  |
7 |     SYN,
  |     ^^^

warning: variant is never constructed: `ACK`
 --> src/flow/flag.rs:8:5
  |
8 |     ACK,
  |     ^^^

warning: variant is never constructed: `CWR`
 --> src/flow/flag.rs:9:5
  |
9 |     CWR,
  |     ^^^

warning: variant is never constructed: `RST`
  --> src/flow/flag.rs:10:5
   |
10 |     RST,
   |     ^^^

warning: associated function is never used: `new`
  --> src/flow/statistic.rs:17:12
   |
17 |     pub fn new(data: &Vec<f64>) -> Self {
   |            ^^^

warning: 9 warnings emitted

warning: variant is never constructed: `FIN`
 --> src/flow/flag.rs:3:5
  |
3 |     FIN,
  |     ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: variant is never constructed: `PSH`
 --> src/flow/flag.rs:4:5
  |
4 |     PSH,
  |     ^^^

warning: variant is never constructed: `URG`
 --> src/flow/flag.rs:5:5
  |
5 |     URG,
  |     ^^^

warning: variant is never constructed: `ECE`
 --> src/flow/flag.rs:6:5
  |
6 |     ECE,
  |     ^^^

warning: variant is never constructed: `SYN`
 --> src/flow/flag.rs:7:5
  |
7 |     SYN,
  |     ^^^

warning: variant is never constructed: `ACK`
 --> src/flow/flag.rs:8:5
  |
8 |     ACK,
  |     ^^^

warning: variant is never constructed: `CWR`
 --> src/flow/flag.rs:9:5
  |
9 |     CWR,
  |     ^^^

warning: variant is never constructed: `RST`
  --> src/flow/flag.rs:10:5
   |
10 |     RST,
   |     ^^^

warning: 8 warnings emitted

    Finished test [unoptimized + debuginfo] target(s) in 0.02s
     Running target/debug/deps/libflow-dadee8299390e67f
thread 'flow::flow_id::tests::test_bad_dst_new' panicked at 'called `Result::unwrap()` on an `Err` value: AddrParseError(())', src/flow/flow_id.rs:84:49
stack backtrace:
   0: rust_begin_unwind
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475
   1: core::panicking::panic_fmt
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/panicking.rs:85
   2: core::option::expect_none_failed
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/option.rs:1221
   3: core::result::Result<T,E>::unwrap
             at /home/kvmk8371/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:973
   4: libflow::flow::flow_id::FlowId::new
             at ./src/flow/flow_id.rs:84
   5: libflow::flow::flow_id::tests::test_bad_dst_new
             at ./src/flow/flow_id.rs:149
   6: libflow::flow::flow_id::tests::test_bad_dst_new::{{closure}}
             at ./src/flow/flow_id.rs:148
   7: core::ops::function::FnOnce::call_once
             at /home/kvmk8371/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227
   8: core::ops::function::FnOnce::call_once
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/ops/function.rs:227
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'flow::flow_id::tests::test_bad_src_new' panicked at 'called `Result::unwrap()` on an `Err` value: AddrParseError(())', src/flow/flow_id.rs:83:49
stack backtrace:
   0: rust_begin_unwind
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475
   1: core::panicking::panic_fmt
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/panicking.rs:85
   2: core::option::expect_none_failed
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/option.rs:1221
   3: core::result::Result<T,E>::unwrap
             at /home/kvmk8371/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:973
   4: libflow::flow::flow_id::FlowId::new
             at ./src/flow/flow_id.rs:83
   5: libflow::flow::flow_id::tests::test_bad_src_new
             at ./src/flow/flow_id.rs:137
   6: libflow::flow::flow_id::tests::test_bad_src_new::{{closure}}
             at ./src/flow/flow_id.rs:136
   7: core::ops::function::FnOnce::call_once
             at /home/kvmk8371/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:227
   8: core::ops::function::FnOnce::call_once
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/ops/function.rs:227
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
   Doc-tests libflow

Process finished with exit code 0
```
